constantdict/__init__.py

```diff
@@ -170,3 +170,36 @@ def discard(self, key: K) -> constantdict[K, V]:
     setdefault = _del_attr  # type: ignore[assignment]
 
     # }}}
+
+    # {{{ mutation
+
+    def mutate(self) -> constantmutabledict[K, V]:
+        """Return a mutable version of this :class:`constantdict`."""
+        # Based on the immutables.Map API.
+        # This needs to make a copy since the original dictionary must
+        # not be modified.
+        return constantmutabledict(self)
+
+    # }}}
+
+
+class constantmutabledict(constantdict[K, V]):  # noqa: N801
+    """A mutable immutable dictionary."""
```

Comment:
This is a bit nonsensical. I'd come up with better naming.

Reply:
What do you think of 2e054bd2b3614c3863e26cab833c056aeeba8e05?